src/realm/object-store/results.cpp

```diff
@@ -316,6 +316,24 @@ util::Optional<Obj> Results::try_get(size_t row_ndx)
     return util::none;
 }
 
+Mixed Results::get_any(size_t ndx)
+{
+    util::CheckedUniqueLock lock(m_mutex);
+    validate_read();
+    if (m_mode == Mode::List) {
```

Comment:
This needs to handle the other results modes, similar to what try_get does above it.

Reply:
The function above is used from get<Obj>. That function does only make sense for the modes where the result object contains Objects. The function get_any only makes sense when result represents a list. Compare with function starting line 238.

Reply:
Why can't get_any be used when the results contains objects?

Reply:
Because Mixed can only hold primitive types.